src/features/pages/home/live-pricing/table-component/live-market-table.tsx

```diff
@@ -10,58 +10,33 @@ import { TAvailableLiveMarkets, TMarketData } from '../types'
 import useLiveColumns from '../use-live-columns'
 import { table_row_header, table_row_data } from './live-pricing.module.scss'
 import Flex from 'features/components/atoms/flex-box'
-import useRegion from 'components/hooks/use-region'
 import Link from 'features/components/atoms/link'
 import { Localize } from 'components/localization'
+import useFetchFirebaseData from 'components/hooks/use-firebase-fetch-data'
 
 export type TLiveMarketTableProps = {
     selected_market: TAvailableLiveMarkets
     link_to: string
 }
 
 const LiveMarketTable = ({ selected_market, link_to }: TLiveMarketTableProps) => {
-    const { is_eu } = useRegion()
-    const [initial_loaded, setInitialLoaded] = useState(false)
+    const firebaseData = useFetchFirebaseData()
 
-    const [rawMarketsData, setRawMarketsData] = useState()
+    const rawMarketsData = firebaseData.data
     const TABLE_VISIBLE_ROWS = 5
     const [markets_data, setMarketsData] = useState(() => {
         const temp = new Map<TAvailableLiveMarkets, TMarketData[]>()
         return temp
     })
-    const intervalRef = useRef(null)
 
     const [sorting, setSorting] = React.useState<SortingState>([])
 
-    const region = useMemo(() => {
-        return is_eu
-            ? 'https://deriv-static-pricingfeed.firebaseio.com/eu.json'
-            : 'https://deriv-static-pricingfeed.firebaseio.com/row.json'
-    }, [is_eu])
-
-    const getData = useCallback(async () => {
-        const rawResponse = await fetch(region)
-        const response = await rawResponse.json()
-        setRawMarketsData(response)
-    }, [region])
-
-    useEffect(() => {
-        if (!initial_loaded) {
-            getData()
-            setInitialLoaded(true)
-        }
-        intervalRef.current = setInterval(getData, 10000)
-        return () => {
-            clearInterval(intervalRef.current)
-        }
-    }, [getData, initial_loaded])
-
     const updateData = useCallback(() => {
```

Comment:
I don't think we need to do this as a callback any more.

Maybe instead the mutated data can be a `useMemo`?

Then we can remove these lines from below:

```javascript
    useEffect(() => {
        updateData()
    }, [updateData])
```

Reply:
Yeah that's right useMemo works for us, but to load the function and not have multiple render kept the useEffect hook

Reply:
Hmm, do you think it needs to be a function though. It seems like the only output of that `updateData` function is to run `setMarketsData`, therefore setting `marketsData` to the new value.

I was wondering, can we not just do `const marketsData = useMemo(() => {...`